src/DataDeps.jl

```diff
@@ -26,8 +26,12 @@ struct DataDep{H, R, F, P} <: AbstractDataDep
     extra_message::String
 end
 
-macro datadep_str(name)
-    :(resolve(registry[$(esc(name))], @__FILE__))
+macro datadep_str(path)
+    parts = splitpath(path)
+    name = first(parts)
+    inner_path = length(parts) > 1 ? joinpath(Iterators.drop(parts, 1)...) : ""
+
+    :(resolve(registry[$(esc(name))], @__FILE__, $(esc(inner_path))))
```

Comment:
does this change break the option to use `@datadep_str myvariable` ? i.e. using a not hardcoded string.

i do currently use this here https://github.com/JuliaML/MLDatasets.jl/blob/datadeps/src/io/download.jl#L17 , but I think this may go away will all the new changes that you introduced

Reply:
Yes, it does.
I didn't realize that worked at all.
I should make it continue working.